distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso

```diff
@@ -155,30 +152,23 @@ type Statistic
        - statistics: Set of statistics to calculate.
     compute_bulk : Vector -> Vector Statistic -> Vector Any
     compute_bulk data statistics=[Statistic.Count, Statistic.Sum] =
-        counter = CountMinMax.new
-
         moment_order = statistics.map .order
-        skip_moments = moment_order.all s->s.is_nothing
-        moments = if skip_moments then Nothing else Moments.new (moment_order.filter (v-> v != Nothing) . fold 0 .max)
-
-        result = data.fold moments current-> value-> compute_fold counter current value
+        max_moment_order = moment_order.filter (v-> v != Nothing) . fold 0 .max
+        counter = data.fold (Accumulator.new max_moment_order) current-> value-> compute_fold current value
 
         report_error stats =
             first = 0.up_to stats.length . find i-> (moment_order.at i . is_nothing . not)
             stat = stats.at first
             Error.throw (Illegal_Argument.Error ("Can only compute " + stat.to_text + " on numerical data sets."))
 
         if counter.count == 0 then statistics.map empty_value else
-            if skip_moments.not && result.is_nothing then report_error statistics else
+            if max_moment_order > 0 && counter.moments.is_nothing then report_error statistics else
                 statistics.map statistic-> case statistic of
-                    Statistic.Count -> counter.count
-                    Statistic.Minimum -> if counter.comparatorError then (Error.throw (Incomparable_Values.Error Nothing Nothing)) else counter.minimum
-                    Statistic.Maximum -> if counter.comparatorError then (Error.throw (Incomparable_Values.Error Nothing Nothing)) else counter.maximum
                     Statistic.Covariance series -> calculate_correlation_statistics data series . covariance
                     Statistic.Pearson series -> calculate_correlation_statistics data series . pearsonCorrelation
                     Statistic.Spearman series -> calculate_spearman_rank data series
                     Statistic.R_Squared series -> calculate_correlation_statistics data series . rSquared
```

Comment:
While this is not a regression since that part of the code is unchanged, I have only now noticed that in each branch `Covariance | Pearson | R_Squared` we call `calculate_correlation_statistics` that is doing a pretty heavy computation.

I assume the whole idea behind `compute_bulk` was to avoid recomputing the same moments for related statistics on the same data. It seems that the same principle can be applied here - we could check if _any_ of these 3 is needed and compute the `calculate_correlation_statistics` only once instead of 1-3 times.

Reply:
The logic is that the series do not necessarily refer to the same second series.

We can now compute a set of unique second series and then compute the limited set of correlation stats but viewed that this was too complicated for a small scenario.

Reply:
Oh! I have missed that part. Then totally seems sensible to keep it as is for now.